### DISCUSSION TITLE
How to resolve includePath conflict?

### DISCUSSION THREAD

**Discussion by bisbaldk** [microsoft/vscode-cpptools#7095] [Q&A] [answered]
I have a workspacefolder that contains multiple sub-dirs,
I'm using `"${workspaceFolder}/**"` for recursive inclusion, each of the sub-dirs in this workspacefolder has a folder called include, 
now I have 2 header files that have the same name but different API signatures, 
one cpp file seems to include the wrong header file, it tries to resolve the header file that's not in its current folder, which has different APIs, and thus it gives a bunch of error squiggles, how do I resolve this?

To make this simpler, I have `A.cpp`, its path is `/lab1/src/A.cpp`, I want it to include `/lab1/include/A.hpp`, but now it tries to resolve `/lab2/include/A.hpp`, in lab2, `A.hpp` has different signatures, so it can't resolve them..

**Comment by sean-mcmanus** [comment, marked as answer]
One potential solution is to add a non-recursive include path, such as /lab2/include which will force that path to always get priority, and potentially use 2 different configurations to switch which one gets priority. Another is to use to use a workspace with multiple workspace folders. Another solution is to add the folders you don't want getting used to the files.exclude setting. Another solution is to set the compileCommands or configurationProvider settings (CMake Tools, Makefile Tools, etc.).

**Reply by bisbaldk** [reply]
I've thought about separating them into different workspaces, which is probably the most ideal solution for now, but wouldn't there be a more elegant solution to it? Since this big folder that contains both /lab1 and /lab2 is my current workspace folder, I just pull everything from my github repo, if I need to separate them, then I'll have to manually do the separation each time, I wonder is there a way to assign configuration detail to a single folder that's in the workspace folder?
In this case I have monorepo as the workspace folder, the structure goes like:
`monorepo/lab1`, `monorepo/lab2`, is there a way to specify different configuration for lab1 and lab2?

Also I don't think hardcoded non-recursive approach would be practical, for example if I specify the include path to be `monorepo/lab1/include`, and if I want to do something with `lab2`, it will try to resolve the header file in`lab1`, so I'd have to do this configuration thing over and over again, which could be pretty clunky..

**Reply by sean-mcmanus** [reply]
We don't support nested c_cpp_properties.json in subfolders in a single workspace folder (there was a feature request on it, but I'm not able to locate it). You'd have to create a .code-workspace with the 2 folders in order for the different c_cpp_properties.json to be used in the different folders.
